src/legacy/core_plugins/data/public/search/search_source/search_source.ts

```diff
@@ -192,21 +199,22 @@ export class SearchSource {
    * @async
    */
   async fetch(options: FetchOptions = {}) {
-    const $injector = await chrome.dangerouslyGetActiveInjector();
-    const es = $injector.get('es') as ApiCaller;
+    const searchService = getSearchService();
+    const config = getUiSettings();
 
     await this.requestIsStarting(options);
 
     const searchRequest = await this.flatten();
     this.history = [searchRequest];
 
+    const esShardTimeout = getInjectedMetadata().getInjectedVar('esShardTimeout') as number;
```

Comment:
I think we can actually get rid of this, because it's injected by the search service itself:

https://github.com/elastic/kibana/blob/5853360d75ee17cd00c777d2b2c28fd2f8abd727/src/plugins/data/server/search/es_search/es_search_strategy.ts#L34-L36

In addition, we can remove each of those params (including `preference` since it's injected here:

https://github.com/elastic/kibana/blob/5853360d75ee17cd00c777d2b2c28fd2f8abd727/src/plugins/data/public/search/es_search/es_search_strategy.ts#L35

These can be removed from this portion:

https://github.com/elastic/kibana/blob/5853360d75ee17cd00c777d2b2c28fd2f8abd727/src/legacy/core_plugins/data/public/search/fetch/get_search_params.ts#L34-L39

But we'll need to keep `max_concurrent_shard_requests` and `ignore_throttled` since they're not currently being handled by the search service.

Reply:
I removed the following from `getSearchParams`
 - `rest_total_hits_as_int`
 - `ignore_unavailable`
 - `preference`
 - `timeout`

However, since `msearch` still needs `esShardTimeout`, I can't remove it.